solutions/CSES Increasing Subsequence II.mdx

```diff
@@ -2,10 +2,16 @@
 id: cses-1748
 source: CSES
 title: Increasing Subsequence II
-author: Benjamin Qi, Óscar Garries
+author: Benjamin Qi, Óscar Garries, Timothy Gao
 ---
 
-Coordinate compress, then range sum.
+Let $dp(i)$ be the number of increasing subsequences ending on the right at $i$.
+Notice that $dp(i)$ can be calculated as the summation of all $dp(j)$ such that $j$ is in the range $[0,i]$ and nums[$j$] < nums[$i$].
+
+Naively, this gives an $O(N^2)$ solution. In order to speed this up, we can
+coordinate compress, then range sum.
+
+Particularly, we sort all numbers in nums and map each distinct number to its index in the sorted array, then create a BIT based on these numbers. Now, when we query, we simply find the summation of all $j<k$ in the BIT, where $k$ is the index nums[$i$] maps to. 
```

Comment:
I don't think "particularly" is the right word to use here. Something like "More specifically" maybe?

Also, that's not exactly how the BIT works. I think you should mention how you update it as you sweep through the array. (Right now, it sounds like you initialize the BIT with all values already instead of all 0s initially)

Reply:
Right. "More specifically" sounds like a better word choice. I'll change that.

I wasn't being very clear with my explanation on the BIT, I'll update that. I kind of assumed that the reader already had knowledge of using a range query structure for the LIS problem which on hindsight I should not do.